recite/admin.py

```diff
@@ -55,89 +69,91 @@ class ReciterAdmin(admin.ModelAdmin):
     def save_model(self, request, obj, form, change):
         super().save_model(request, obj, form, change)
 
-        # process csv file uploaded
-        segments_file = form.cleaned_data['segments_file'].file
-        csv_content = self.read_segments_csv(
-            segments_file, encoding=request.encoding)
+        # Create dict segments_dict[surah][ayah] = segments
+        segments_dict = self.get_segments_dict(request=request, form=form)
 
-        # process audio zip file uploaded
+        # Extract audio files zip to temp directory
         audio_zip_file = form.cleaned_data['audio_zip_file']
-        temp_dir = self.get_unzipped_files_dir(audio_zip_file)
-        file_paths = self.get_file_paths(temp_dir)
-
-        # create Recitation objects from processed data
-        for row in csv_content:
-            segments_list = json.loads(row.get("segments"))
-            segments = ",".join(
-                (f"{segment[2]}:{segment[3]}" for segment in segments_list)
-            )
-
-            ayah = self.get_ayah_object_from_row(row)
-            file_ayah_path = self.get_file_audio_path(file_paths, ayah)
-
-            with open(file_ayah_path, 'rb') as f:
-                file_ayah = File(f)
-
-                recitation, _ = Recitation.objects.get_or_create(
-                    ayah=ayah,
-                    reciter=obj,
-                    segments=segments,
-                    audio=file_ayah
-                )
-
-        shutil.rmtree(temp_dir)
+        audio_files = ZipFile(audio_zip_file, 'r')
 
-    def get_file_paths(self, temp_dir):
-        """Return file paths from temp_dir checking subdirectories"""
-        file_paths = []
-        for root, directories, files in os.walk(temp_dir):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            audio_files.extractall(temp_dir)
+
+            # let's create dict file_paths[surah][ayah] = file_path
+            file_paths = self.get_file_paths_dict(temp_dir=temp_dir)
+            # let's create Recitations from segments and file_paths
+            for surah_number, ayahs in segments_dict.items():
+                for ayah_number, segments in ayahs.items():
+                    with open(file_paths[surah_number][ayah_number], 'rb')\
+                            as audio_file:
+
+                        file_ayah = File(audio_file)
+                        ayah = Ayah.objects.get(
+                            number=ayah_number,
+                            surah=surah_number,
+                        )
+
+                        Recitation.objects.get_or_create(
+                            ayah=ayah,
+                            reciter=obj,
+                            segments=segments,
+                            audio=file_ayah
+                        )
+
+    def get_file_paths_dict(self, temp_dir):
```

Comment:
Nit: it doesn't matter, is it a temporary directory or not. Call it just dir or directory